lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                    quarantine(qdir, idxctx.logger, tb)
+                    # Audit should pick up missing .md5 file in ARCHIVE directory.
+                    continue
+                else:
+                    tarballs.append((size, controller, tb))
+        except SigTermException:
+            # Re-raise a SIGTERM to avoid it being lumped in with general
+            # exception handling below.
+            raise
+        except Exception:
+            idxctx.logger.exception(
+                "Unexpected error encountered generating list"
+                " of tar balls to process"
+            )
+            # tuple to return the status and return value
+            return (12, [])
+        else:
+            if not tarballs:
+                idxctx.logger.info("No tar balls found that need processing")
+                return (0, [])
+
+        return (0, sorted(tarballs))
+
+    def process_tb(self, tarballs):
+        """Process Tarballs For Indexing and creates report
+
+            "tarballs" - List of tarball, it is the second value of
+                the tuple returned by collect_tb() """
+
+        # We always process the smallest tar balls first.
+        idxctx = self.idxctx
+
+        # At this point, tarballs contains a list of tar balls sorted by size
+        # that were available as symlinks in the various 'linksrc' directories.
+        idxctx.logger.debug("Preparing to index {:d} tar balls", len(tarballs))
+
+        try:
+            # Now that we are ready to begin the actual indexing step, ensure we
+            # have the proper index templates in place.
+            idxctx.logger.debug("update_templates [start]")
+            idxctx.templates.update_templates(idxctx.es)
+        except TemplateError as e:
+            idxctx.logger.error("update_templates [end], error {}", repr(e))
+            res = 9
+        except SigTermException:
+            # Re-raise a SIGTERM to avoid it being lumped in with general
+            # exception handling below.
+            raise
+        except Exception:
+            idxctx.logger.exception(
+                "update_templates [end]: Unexpected template" " processing error"
+            )
+            res = 12
+        else:
+            idxctx.logger.debug("update_templates [end]")
+            res = 0
+
+        if res != 0:
+            # Exit early if we encounter any errors.
+            return res
+
+        report = Report(
+            idxctx.config,
+            self.name,
+            es=idxctx.es,
+            pid=idxctx.getpid(),
+            group_id=idxctx.getgid(),
+            user_id=idxctx.getuid(),
+            hostname=idxctx.gethostname(),
+            version=VERSION,
+            templates=idxctx.templates,
+        )
+        # We use the "start" report ID as the tracking ID for all indexed
+        # documents.
+        try:
+            tracking_id = report.post_status(tstos(idxctx.time()), "start")
+        except SigTermException:
```

Comment:
Do we need to handle SigTermException here?

Reply:
yes, to not get confused with general exception

Reply:
@ndokos, see the comment on lines 185 and 186. If there is a better way to word that comment to make this requirement clear, let me know.